src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -780,11 +783,22 @@ private int WritePosixAndGnuSharedFields(Span<byte> buffer)
             return checksum;
         }
 
+        private int WriteAsGnuTimestamp(DateTimeOffset timestamp, Span<byte> buffer)
+        {
+            if (_typeFlag is TarEntryType.LongLink or TarEntryType.LongPath || timestamp == DateTimeOffset.UnixEpoch)
+            {
+                buffer.Clear();
```

Comment:
There are various methods in this file that assume the buffer to be zero-ed out beforehand. So the call to `buffer.Clear()` should not be needed.

I assume the intent of this change is to not write dates for TarEntryType.LongLink and TarEntryType.LongPath. Instead of introducing `WriteAsGnuTimestamp`, consider moving it to the caller:

```cs
private int WriteGnuFields(Span<byte> buffer)
{
    int checksum = 0;
    if (_typeFlag is not TarEntryType.LongLink and not TarEntryType.LongPath)
    {
      checksum += WriteAsTimestamp(_aTime, buffer.Slice(FieldLocations.ATime, FieldLengths.ATime));
      checksum += WriteAsTimestamp(_cTime, buffer.Slice(FieldLocations.CTime, FieldLengths.CTime));
    }
    ...
```

Reply:
I forgot to answer to this comment, which happened before [the above discussion](https://github.com/dotnet/runtime/pull/115211#discussion_r2071117749) and clarification of the intent of this PR: Which is to not write ctime and atime anywhere.